### PR TITLE
Hotfix staking maps

### PR DESCRIPTION
**Pull request by unconst**
## Description
- Adds a call to perform a hotfix which swaps over already-swapped scheduled coldkeys' delegated stake, as intended. 
    - Also updates StakingHotkeys map and corresponding stake values. 
    - Adds corresponding tests
- Patches the coldkey stake swap function to include delegated stake entries and map entries.
- Removes hotkey swap call (temporary)

### CODE REVIEW

File: pallets/subtensor/src/swap.rs [hunk 1]
```diff
@@ -907,6 +952,39 @@ impl<T: Config> Pallet<T> {
             TotalColdkeyStake::<T>::get(new_coldkey)
         );
     }
+
+
+	pub fn swap_hotfix(old_coldkey: &T::AccountId, new_coldkey: &T::AccountId) {
+
+		let weight = T::DbWeight::get().reads_writes(2, 1);
```
Comment (sam0x17) [inline pallets/subtensor/src/swap.rs]: this variable is unused

File: pallets/subtensor/src/swap.rs [hunk 2]
```diff
@@ -851,7 +851,8 @@ impl<T: Config> Pallet<T> {
             log::info!("Transferring stake for hotkey {:?}: {}", hotkey, stake);
             if stake > 0 {
                 // Insert the stake for the hotkey and new coldkey
-                Stake::<T>::insert(hotkey, new_coldkey, stake);
+                let old_stake = Stake::<T>::get(hotkey, new_coldkey);
+                Stake::<T>::insert(hotkey, new_coldkey, stake.saturating_add(old_stake));
```
Comment (open-junius) [inline pallets/subtensor/src/swap.rs:855]: can use mutate to add stake

File: pallets/subtensor/src/swap.rs [hunk 3]
```diff
@@ -861,6 +862,52 @@ impl<T: Config> Pallet<T> {
...
+            old_coldkey
+        );
+
+        for staking_hotkey in StakingHotkeys::<T>::get(old_coldkey) {
+            log::info!("Processing staking hotkey: {:?}", staking_hotkey);
+            if Stake::<T>::contains_key(staking_hotkey.clone(), old_coldkey) {
+                let hotkey = &staking_hotkey;
+                // Retrieve and remove the stake associated with the hotkey and old coldkey
+                let stake: u64 = Stake::<T>::get(hotkey, old_coldkey);
+                Stake::<T>::remove(hotkey, old_coldkey);
```
Comment (open-junius) [inline pallets/subtensor/src/swap.rs:876]: no weight for remove.

Comment (open-junius) [inline pallets/subtensor/src/swap.rs:876]: no weight for get and remove.

**Review by camfairchild — DISMISSED** [review]

**Review by distributedstatemachine — DISMISSED** [review]

**Comment by sam0x17** [comment]
that unused weight variable seems to be the only remaining thing, I'll just comment it out

**Comment by camfairchild** [comment]
Included in #636

**Pull request closed without merging**